pandas/core/indexes/datetimes.py

```diff
@@ -1937,16 +1937,21 @@ def tz_convert(self, tz):
                      mapping={True: 'infer', False: 'raise'})
     def tz_localize(self, tz, ambiguous='raise', errors='raise'):
         """
-        Localize tz-naive DatetimeIndex to given time zone (using
-        pytz/dateutil), or remove timezone from tz-aware DatetimeIndex
+        Localize tz-naive DatetimeIndex to tz-aware DatetimeIndex.
+
+        This method takes a naive DatetimeIndex object and interprets it as
+        if it is in that timezone. It does not move the time to another
+        timezone.
+        Time zone localization helps to switch b/w time zone aware and time
+        zone unaware objects.
 
         Parameters
         ----------
         tz : string, pytz.timezone, dateutil.tz.tzfile or None
             Time zone for time. Corresponding timestamps would be converted to
             time zone of the TimeSeries.
             None will remove timezone holding local time.
-        ambiguous : 'infer', bool-ndarray, 'NaT', default 'raise'
+        ambiguous : {'infer', 'NaT', default 'raise'}, bool-ndarray
```

Comment:
I would make this `str {'infer', 'NaT', 'raise'} or boolean array, default 'raise'`

But that is probably too long, so you can do:

```
        ambiguous : str {'infer', 'NaT', 'raise'} or bool array, \
default 'raise'
            - 'infer': ...
```

It is a bit ugly in the code, but will render OK in plain text docstring and in html docstring.

Reply:
A little bit below for the "errors" keyword, can you make the type description there: `errors : {'raise', 'coerce'}, default 'raise'` (so with curly braces)

Reply:
errors : {'raise', 'coerce'}, default 'raise' 
we have to define str{'raise', 'coerce'} I thought its already understood

Reply:
Yes, in this case `errors : {'raise', 'coerce'}, default 'raise'` is good (without the 'str')